lib/journaled/outbox/batch_processor.rb

```diff
@@ -0,0 +1,146 @@
+# frozen_string_literal: true
+
+module Journaled
+  module Outbox
+    # Processes batches of outbox events
+    #
+    # This class handles the core business logic of:
+    # - Fetching events from the database (with FOR UPDATE)
+    # - Sending them to Kinesis via batch sender
+    # - Handling successful deliveries (deleting events)
+    # - Handling failures (incrementing attempts, marking as failed if max reached)
+    #
+    # All operations happen within a single database transaction for consistency.
+    # The Worker class delegates to this for actual event processing.
+    class BatchProcessor
+      def initialize
+        @batch_sender = KinesisBatchSender.new
+      end
+
+      # Process a single batch of events
+      #
+      # Wraps the entire batch processing in a single transaction:
+      # 1. SELECT FOR UPDATE (claim events)
+      # 2. Send to Kinesis
+      # 3. Delete successes
+      # 4. Increment attempts and/or mark failures
+      #
+      # @return [Hash] Statistics with :succeeded, :failed_transiently, :failed_permanently counts
+      def process_batch
+        ActiveRecord::Base.transaction do
+          # Fetch and lock events within transaction
+          events = Event.fetch_batch_for_update
+
+          Rails.logger.info("Processing batch of #{events.count} events")
```

Comment:
```suggestion
          Rails.logger.info("[journaled] Processing batch of #{events.count} events")
```

Or maybe we can wrap the processor's loop with automatic log tagging -- something to indicate that these log lines are coming from the `journaled` gem.